solr/modules/ltr/src/java/org/apache/solr/ltr/LTRRescorer.java

```diff
@@ -233,7 +235,14 @@ protected static boolean scoreSingleHit(
     boolean logHit = false;
 
     scorer.getDocInfo().setOriginalDocScore(hit.score);
+    QueryLimits queryLimits = QueryLimits.getCurrentLimits();
```

Comment:
IMO it's better do do such checks at the spot where there is a loop over something like docs.  This also allows fetching the QueryLimits outside the loop.  Thus I suggest modifying `scoreFeatures`.  You could do the `queryLimits.maybeExitWithPartialResults` check at the start of the `while` loop.

Reply:
Ah; I see you do this here so that LTRInterleavingRescorer will incorporate this, which overrides `scoreFeatures` completely yet calls `scoreSingleHit`.  Okay fine.   But note `scoreFeatures` has lots of code duplication; it'd benefit from a refactor.

If you want to keep queryLimits check in `scoreSingleHit` then you needn't declare queryLimits to only then call a method on it 2 lines later (kinda weird; why score in-between?).  Just do `QueryLimits.getCurrentLimits().maybeExitWithPartialResults(...)`

Reply:
Yes David you are correct about the reason and I agree scoreFeatures would deserve some care, but I prefer to dedicate a separate PR to that.

Your observation about the declare and then call is also welcome, the change is coming in my next commit.
Not sure why I did it in that way, maybe a copy and paste from a previous version of the code outside the loop,

Reply:
> ... scoreFeatures would deserve some care, but I prefer to dedicate a separate PR to that. ...

+1 to not mixing code changes with code care.

And I vaguely recall something around this previously, possibly the #478 (now stale) draft.